jfr-daemon/src/main/java/org/slf4j/impl/StaticLoggerBinder.java

```diff
@@ -0,0 +1,236 @@
+package org.slf4j.impl;
+
+import com.newrelic.api.agent.Agent;
+import com.newrelic.api.agent.Logger;
+import com.newrelic.api.agent.NewRelic;
+import java.util.logging.Level;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.helpers.FormattingTuple;
+import org.slf4j.helpers.MarkerIgnoringBase;
+import org.slf4j.helpers.MessageFormatter;
+import org.slf4j.spi.LoggerFactoryBinder;
+
+public class StaticLoggerBinder implements LoggerFactoryBinder {
+
+  private static final StaticLoggerBinder SINGLETON = new StaticLoggerBinder();
+  private static final String LOGGER_NAME = "com.newrelic.jfr.logger";
+
+  public static StaticLoggerBinder getSingleton() {
+    return SINGLETON;
+  }
+
+  public static String REQUESTED_API_VERSION = "1.6.99";
+
+  private static final String loggerFactoryClassStr = AgentLoggerFactory.class.getName();
+
+  private final ILoggerFactory loggerFactory;
+
+  private StaticLoggerBinder() {
+    Agent newRelicAgent = NewRelic.getAgent();
+    if (!newRelicAgent.getClass().getName().contains("NoOpAgent")) {
```

Comment:
This is how the daemon decides if it is running embedded in the Java Agent or not.